src/vs/editor/common/controller/cursorCollection.ts

```diff
@@ -167,12 +137,12 @@ export class CursorCollection {
 	}
 
 	private _addSecondaryCursor(): void {
-		this.secondaryCursors.push(new Cursor(this.context));
-		this.lastAddedCursorIndex = this.secondaryCursors.length;
+		this.cursors.push(new Cursor(this.context));
+		this.lastAddedCursorIndex = this.cursors.length - 1;
```

Comment:
I think this is also incorrect

Reply:
I'm not so sure this is incorrect (as well as for your other correctness-comments).

`lastAddedCursorIndex` is still in terms of `secondaryCursors`. The value should still be the same as before.
Because `this.cursors[0]` now is the primary cursor, what used to be `this.secondaryCursors.length` is now `this.cursors.length - 1`.
What used to be `this.secondaryCursors[i]` is now `this.cursors[i + 1]`.

Reply:
:+1: Unfortunately I am unable to follow the proposed change. Please go ahead and merge if you are confident this is correct.